---
title: "Inferring_ASVs_Lake_Michigan"
output: html_document
date: "`r Sys.Date()`"
editor_options: 
  chunk_output_type: console
---
```{r setting-seed}
set.seed(033094)
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r load-libraries}
pacman::p_load(tidyverse, devtools, dada2, phyloseq, patchwork, DT, 
               install = FALSE)
```

## Loading data
```{r load-data}

# Set the raw fastq path to the raw sequencing files 
# Path to the fastq files 
raw_fastqs_path <- "data/01_DADA2/01_raw_gzipped_fastqs"
raw_fastqs_path

# Intuition check
head(list.files(raw_fastqs_path))

#Looking at how many files there are

str(list.files(raw_fastqs_path))

# Create vector of forward reads
forward_reads <- list.files(raw_fastqs_path, pattern = "R1_001.fastq.gz", full.names = TRUE)  
# Intuition Check 
head(forward_reads)  

# Create a vector of reverse reads 
reverse_reads <- list.files(raw_fastqs_path, pattern = "R2_001.fastq.gz", full.names = TRUE)
head(reverse_reads)

```

## Initial Raw Read Quality Assessment

```{r raw-read-assessment}

#Randomly selecting 12 samples from the dataset
random_samples <- sample(1:length(forward_reads), size = 12)
random_samples

#Plotting quality of forward and reverse reads for randomly chosen samples

forward_rawQual_plot_12 <- plotQualityProfile(forward_reads[random_samples]) +  labs(title = "Forward Read Raw Quality")



reverse_rawQual_plot_12 <- plotQualityProfile(reverse_reads[random_samples]) + 
  labs(title = "Reverse Read Raw Quality")


grid.arrange(forward_rawQual_plot_12, reverse_rawQual_plot_12, ncol = 2)

```

```{r aggregating-all-QC-plots}

# Aggregate all QC plots 
plotQualityProfile(forward_reads, aggregate = TRUE) + 
  plotQualityProfile(forward_reads, aggregate = TRUE)

```

