---
title: "Phyloseq PreProcessing"
author: "Daniyal Tariq"
date: "2024-03-18"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


## Setup
```{r setup}
knitr::opts_chunk$set(echo = TRUE,
                      fig.align = "center",
                      fig.path = "../figures/02_PreProcessing/") # send any figure output to this folder
```

## Goals

In this document, we will take the data files generated from the DADA2 workflow and create a phyloseq object. 

Before we do this we will need to perform some preprocessing on the following files:
1. ASV Table
2. Taxonomy Table
3. Metadata

We will then remove mitochondrial and chloroplast reads.

## Loading Relevant Packages
```{r loading packages}
library(devtools)
#devtools::install_github("joey711/phyloseq")
library(phyloseq)
library(tidyverse)
```

## Loading Data

## ASV Table
```{r loading-asv-counts}
# First, load asv table
load("data/02_DADA2/ASV_counts.RData")

```

# Fixing Column Names
```{r fixing-asv-tab-colnames}
# Inspect asv_tab
head(asv_tab)[,1:5]

# Fix names 
sample_names <- colnames(asv_tab)
samples_fixed <- sapply(strsplit(basename(sample_names), "_"), `[`,1) 
head(samples_fixed)

# re-write the ASV count file to fix names 
colnames(asv_tab) <- samples_fixed
str(asv_tab)
```

## Taxonomy Table
```{r loading-taxonomy-table-making-matrix}
tax_df <- read.table("data/02_DADA2/ASV_taxonomy.tsv", sep = "\t", skip = 1)
head(tax_df)

colnames(tax_df) <- c("asv_names", "Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species", "ASV", "ASVseq")

head(tax_df)

# Taxonomy Table Matrix
tax_mat <- 
  tax_df %>%
  tibble::column_to_rownames(., var = "asv_names") %>%
  as.matrix()
```

## Track Reads Data

```{r}
load("data/02_DADA2/track_read_counts.RData")

# Initial data inspection
head(track_counts_df)
dim(track_counts_df)
```

# Inspecting the metadata

```{r}

metadata_df <- read.csv("data/metadata.csv")
dim(metadata_df)
colnames(metadata_df)
# Merge metadata_df with track_reads_df 
head(metadata_df)
head(track_counts_df)
```

# Combining track read counts with metadata
```{r}
metadata_track_reads_df <- 
  metadata_df %>%
  left_join(., track_counts_df, by = "names") 

# Intuition check 
head(metadata_track_reads_df)
```

```{r}
# Update row.names to be sample names 
## Before 
row.names(metadata_track_reads_df)

# Rewrite 
row.names(metadata_track_reads_df) <- metadata_track_reads_df$names
# Check afterwards that it worked 
row.names(metadata_track_reads_df)

# intuition check
head(metadata_track_reads_df)
```

## Handing off our files to phyloseq

# Final check to make sure all objects have the expected dimensions
```{r}
dim(asv_tab)
dim(tax_mat)
# Intuition check 
stopifnot(row.names(asv_tab) == row.names(tax_mat))
```

# Creating the phyloseq object
```{r}
# Construct the phyloseq object 
raw_physeq <- phyloseq(otu_table(asv_tab, taxa_are_rows = TRUE),
                       sample_data(metadata_track_reads_df),
                       tax_table(tax_mat))
raw_physeq
# Save this raw phyloseq object 
save(raw_physeq, file = "data/02_PreProcessing/raw_physeq.RData")
```

# Removing chloroplasts
```{r removing-chloro-asvs}
# Make new physeq object after removing chloroplasts
noChloros_physeq <- 
  raw_physeq %>% 
  # rm chloroplasts
  subset_taxa(Order != "Chloroplast" | is.na(Order))
  
# Checking how many taxa were chloroplasts
num_chloro_ASVs <- ntaxa(raw_physeq) - ntaxa(noChloros_physeq)
num_chloro_ASVs
```

# Removing mitochondria
```{r removing-mito-asvs}
# remove mitochondria 
noChlorosMitos_physeq <- 
  noChloros_physeq %>%
  subset_taxa(Family != "Mitochondria" | is.na(Family))

# Checking how many taxa were mitochondria
num_mito_ASVs <- ntaxa(noChloros_physeq) - ntaxa(noChlorosMitos_physeq)
num_mito_ASVs
```

# Inspecting the phyloseq object after mitochondria and chloroplast removal
```{r inspecting-clean-phyloseq}

noChlorosMitos_physeq

# How many total asvs were removed from chloros and mitos 
ntaxa(raw_physeq) - ntaxa(noChlorosMitos_physeq)

# proportion of asvs kept? 
ntaxa(noChlorosMitos_physeq)/ntaxa(raw_physeq)
```

## Removing the control samples

```{r removing-asvs-in-control-samples}
# create a vector of samples that were negative controls 
control_samples <- c("waterA06222016", "waterB06222016", 
                     "waterC06222016", "waterD06222016","PBSRINSE")

# Make a new phyloseq object of only the control samples 
control_physeq <- noChlorosMitos_physeq %>%  subset_samples(., names %in% control_samples) 

# Looks like all ASVs in the control samples have a zero count. It would appear that we don't need to remove any control ASVs from the true samples.

# Remove the control samples from analysis

noControlsChlorosMitos_physeq <- 
  noChlorosMitos_physeq %>%
  # Remove control samples 
  subset_samples(., !(names %in% control_samples))

# Inspect no controls object 
noControlsChlorosMitos_physeq
```


## Evaluating Mock Community

```{r}

mock_community_sample_names <- c("mockA06222016","mockB06222016", "mockC06222016", "mockD06222016")

mock_physeq <- 
  noControlsChlorosMitos_physeq %>%
  subset_samples(., names %in% mock_community_sample_names)  %>%
  prune_taxa(taxa_sums(.) > 0, .)


# Inspecting mock physeq object
mock_physeq

tax_table(mock_physeq)
cat("DADA2 inferred", ntaxa(mock_physeq), "ASVs present in the mock community.")

#Remove mock communities

noMockControlsChlorosMitos_physeq <- 
  noControlsChlorosMitos_physeq %>%
  # Remove control samples 
  subset_samples(., !(names %in% mock_community_sample_names))

noMockControlsChlorosMitos_physeq
```



```{r}

seqSums_df <- noMockControlsChlorosMitos_physeq %>%
  otu_table() %>%
  # Sum each sample column 
  colSums() %>%
  data.frame() %>%
  rownames_to_column(var = "names") %>%
  left_join(., metadata_track_reads_df, by = "names")

# Rename second column 
colnames(seqSums_df)[2] <- "TotalSeqs"

# check
dim(seqSums_df)


```


```{r}
# Show the depth of samples 
seqSums_df %>%
  dplyr::select(names, TotalSeqs) %>%
  arrange(TotalSeqs) %>%
  head()

# plot it! 
seqSums_df %>%
  ggplot(aes(x=reorder(names, TotalSeqs), y = TotalSeqs,
             fill = lakesite)) + 
  geom_bar(stat = "identity") 

# Density plot 
seqSums_df %>%
  ggplot(aes(TotalSeqs, fill = lakesite)) +
  geom_density(alpha = 0.5) + scale_y_continuous(limits = c(0,1.25e-04))
```

## Evaluate the sequencing depth
```{r}
noMockControlsChlorosMitos_physeq

# What is the library size/sequencing depth for each sample? 
seqSums_df <- 
  noMockControlsChlorosMitos_physeq %>%
  otu_table() %>%
  # Sum each sample column 
  colSums() %>%
  data.frame() %>%
  rownames_to_column(var = "names") %>%
  left_join(., metadata_track_reads_df, by = "names") 

# Rename second column 
colnames(seqSums_df)[2] <- "TotalSeqs"

# check
dim(seqSums_df)

```

```{r}
# Show the depth of samples 
seqSums_df %>%
  dplyr::select(names, TotalSeqs) %>%
  arrange(TotalSeqs) %>%
  head()

# plot it! 
seqSums_df %>%
  ggplot(aes(x=reorder(names, TotalSeqs), y = TotalSeqs,
             fill = lakesite)) + 
  geom_bar(stat = "identity")

# Density plot 
seqSums_df %>%
  ggplot(aes(TotalSeqs, fill = lakesite)) +
  geom_density(alpha = 0.5) + scale_y_continuous(limits = c(0,1.25e-04))
```

#Remove samples with few reads

```{r}
# What's the min seq depth? 
min(sample_sums(noMockControlsChlorosMitos_physeq))

subset_1 <- seqSums_df %>% filter(TotalSeqs < 9) 
names_r <- subset_1$names




```

```{r}
raw_preprocessed_physeq <- 
  noMockControlsChlorosMitos_physeq %>%
  subset_samples(., !(names %in% names_r))

#What's the new min seq depth?
min(sample_sums(raw_preprocessed_physeq))
```


## Save file

```{r}
save(raw_preprocessed_physeq,
     file = "data/02_PreProcessing/raw_preprocessed_physeq.RData")
```

