---
title: "Beta Diversity"
author: "Daniyal Tariq"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##Goals

1.Load in phyloseq data with rooted tree.
2.Evaluate sequencing depth and remove sample.
3.Normalize the read counts between samples.
4.Calculate community dissimilarities. Numbers between 0 and 1. If 0, completely similar versus if they are 1, then theyâ€™re completely dissimilar.
  a.Sorensen: Shared Species as a binary value: Abundance-unweighted
  b.Bray-Curtis: Shared Abundant species: Abundance-weighted
  c.(Abundance-)Weighted UNIFRAC: Consider Abundant Species and where they fall on the tree
5.Visualize the community data with two unconstrained Ordinations:
  a.PCoA: Linear Method. Eigenvalue = how much variation is explained by each axis. Choose to view axis 1, 2, 3, etc. and plot them together.
  b.NMDS: Non-linear. Smush multiple Dimensions into 2 or 3 axes. Need to report Stress value (ideally <0.15).
6.Run statistics with PERMANOVA and betadispR.

##Set the seed

```{r set-seed}
set.seed(2443)
```

##Load Libraries

```{r load-libraries}
#install.packages("vegan")
pacman::p_load(tidyverse, devtools, phyloseq, patchwork, vegan,
               install = FALSE)

lakesite_colors <- c("110" = "dodgerblue4","M15" = "dodgerblue2","M45" = "#D9CC3C","MLB" = "#A0E0BA")
```

##Load Data

```{r load-data}
# Load in rooted phylogenetic tree! 
load("data/03_Phylogenetic_Tree/phytree_preprocessed_physeq.RData")
midroot_physeq
```

##Explore Read Counts

```{r}
# Calculate the total number of reads per sample. 
raw_TotalSeqs_df  <- 
  midroot_physeq %>%
  # calculate the sample read sums 
  sample_sums() %>%
  data.frame()
# name the column 
colnames(raw_TotalSeqs_df)[1] <- "TotalSeqs"
  
head(raw_TotalSeqs_df)
```

```{r}
# make histogram of raw reads 
raw_TotalSeqs_df %>%
  ggplot(aes(x = TotalSeqs)) + 
  geom_histogram(bins = 15) + 
  scale_x_continuous(limits = c(2000, 10000)) + 
  labs(title = "Raw Sequencing Depth Distribution") + 
  theme_classic()
```

```{r}
# what is the minimum number of sequences 
midroot_physeq %>%
  sample_sums() %>%
  min()
```

##Normalize read counts

```{r making-scale-reads-function}

# Function to scale reads: http://deneflab.github.io/MicrobeMiseq/ 
# Scales reads by 
# 1) taking proportions
# 2) multiplying by a given library size of n
# 3) rounding 
# Default for n is the minimum sample size in your library
# Default for round is floor

matround <- function(x){trunc(x+0.5)}

scale_reads <- function(physeq, n = min(sample_sums(physeq)), round = "round") {
  
  # transform counts to n
  physeq.scale <- transform_sample_counts(physeq, function(x) {(n * x/sum(x))})
  
  # Pick the rounding functions
  if (round == "floor"){
    otu_table(physeq.scale) <- floor(otu_table(physeq.scale))
  } else if (round == "round"){
    otu_table(physeq.scale) <- round(otu_table(physeq.scale))
  } else if (round == "matround"){
    otu_table(physeq.scale) <- matround(otu_table(physeq.scale))
  }
  
  # Prune taxa and return new phyloseq object
  physeq.scale <- prune_taxa(taxa_sums(physeq.scale) > 0, physeq.scale)
  return(physeq.scale)
}


```

##Scale the reads and check the distribution of the seq depth

```{r}
min(sample_sums(midroot_physeq))

```

```{r}
# Scale reads by the above function
scaled_rooted_physeq <- midroot_physeq %>% scale_reads(round = "matround")

# Calculate the read depth 
scaled_TotalSeqs_df <- scaled_rooted_physeq %>% sample_sums() %>% data.frame()

colnames(scaled_TotalSeqs_df)[1] <-"TotalSeqs"

# Inspect
head(scaled_TotalSeqs_df)
```

```{r}

# Check the range of the data 
min_seqs <- min(scaled_TotalSeqs_df$TotalSeqs); min_seqs

max_seqs <- max(scaled_TotalSeqs_df$TotalSeqs); max_seqs

max_seqs - min_seqs

```

```{r}
# Plot Histogram 
scaled_TotalSeqs_df %>%
  ggplot(aes(x = TotalSeqs)) + 
  geom_histogram(bins = 50) + 
  scale_x_continuous(limits = c(0, 10000)) + 
  labs(title = "Scaled Sequencing Depth at 2194") + 
  theme_classic()
```

##Calculate & Visualize community dissimilarity

#PCoA

Sorensen

```{r}
# Calculate sorensen dissimilarity: Abundance-unweighted of shared taxa
scaled_soren_pcoa <-  
  ordinate(
    physeq = scaled_rooted_physeq,
    method = "PCoA",
    distance = "bray", binary = TRUE)

#str(scaled_soren_pcoa)

# Plot the ordination 
soren_station_pcoa <- plot_ordination(
  physeq = scaled_rooted_physeq,
  ordination = scaled_soren_pcoa,
  color = "lakesite",
  shape = "season",
  title = "Sorensen PCoA") +
  geom_point(size=5, alpha = 0.5, aes(color = lakesite)) +
  scale_shape_manual(values = c(15,16,17)) +
  scale_color_manual(values = lakesite_colors) + 
  theme_bw()
# Show the plot 
soren_station_pcoa
```

```{r}
# Calculate sorensen dissimilarity: Abundance-unweighted of shared taxa
scaled_soren_pcoa <-  
  ordinate(
    physeq = scaled_rooted_physeq,
    method = "PCoA",
    distance = "bray", binary = TRUE)

#str(scaled_soren_pcoa)

# Plot the ordination 
soren_station_pcoa <- plot_ordination(
  physeq = scaled_rooted_physeq,
  ordination = scaled_soren_pcoa,
  color = "lakesite",
  shape = "limnion",
  title = "Sorensen PCoA") +
  geom_point(size=5, alpha = 0.5, aes(color = lakesite)) +
  scale_shape_manual(values = c(15,16,17)) +
  scale_color_manual(values = lakesite_colors) + 
  theme_bw()
# Show the plot 
soren_station_pcoa
```


Bray

```{r}
# Calculate the BC distance
scaled_BC_pcoa <- 
  ordinate(
    physeq = scaled_rooted_physeq,
    method = "PCoA",
    distance = "bray")

# Plot the PCoA
bray_station_pcoa <- 
  plot_ordination(
    physeq = scaled_rooted_physeq,
    ordination = scaled_BC_pcoa,
    color = "lakesite",
    shape = "nuc_acid_type",
    title = "Bray-Curtis PCoA") +
  geom_point(size=5, alpha = 0.5, aes(color = lakesite)) +
  scale_shape_manual(values = c(15,16,17,18, 19, 20,21)) +
  scale_color_manual(values = c(lakesite_colors)) + 
  theme_bw()
bray_station_pcoa
```

```{r}
# Calculate the BC distance
scaled_BC_pcoa <- 
  ordinate(
    physeq = scaled_rooted_physeq,
    method = "PCoA",
    distance = "bray")

# Plot the PCoA
bray_station_pcoa <- 
  plot_ordination(
    physeq = scaled_rooted_physeq,
    ordination = scaled_BC_pcoa,
    color = "lakesite",
    shape = "season",
    title = "Bray-Curtis PCoA") +
  geom_point(size=5, alpha = 0.5, aes(color = lakesite)) +
  scale_shape_manual(values = c(15,16,17,18, 19, 20,21)) +
  scale_color_manual(values = c(lakesite_colors)) + 
  theme_bw()
bray_station_pcoa
```

```{r}
# Calculate the BC distance
scaled_BC_pcoa <- 
  ordinate(
    physeq = scaled_rooted_physeq,
    method = "PCoA",
    distance = "bray")

# Plot the PCoA
bray_station_pcoa <- 
  plot_ordination(
    physeq = scaled_rooted_physeq,
    ordination = scaled_BC_pcoa,
    color = "lakesite",
    shape = "limnion",
    title = "Bray-Curtis PCoA") +
  geom_point(size=5, alpha = 0.5, aes(color = lakesite)) +
  scale_shape_manual(values = c(15,16,17,18, 19, 20,21)) +
  scale_color_manual(values = c(lakesite_colors)) + 
  theme_bw()
bray_station_pcoa
```


Weighted Unifrac

```{r}
# Calculate the BC distance
scaled_wUNI_pcoa <- 
  ordinate(
    physeq = scaled_rooted_physeq,
    method = "PCoA",
    distance = "wunifrac")

# Plot the PCoA
wUNI_station_pcoa <- 
  plot_ordination(
    physeq = scaled_rooted_physeq,
    ordination = scaled_wUNI_pcoa,
    color = "lakesite",
    shape = "lakesite",
    title = "Weighted Unifrac PCoA") +
  geom_point(size=5, alpha = 0.5, aes(color = lakesite)) +
  scale_shape_manual(values = c(15,16,17,18, 19, 20)) +
  scale_color_manual(values = c(lakesite_colors)) + 
  theme_bw()
wUNI_station_pcoa
```

```{r}
# Calculate the BC distance
scaled_wUNI_pcoa <- 
  ordinate(
    physeq = scaled_rooted_physeq,
    method = "PCoA",
    distance = "wunifrac")

# Plot the PCoA
wUNI_station_pcoa <- 
  plot_ordination(
    physeq = scaled_rooted_physeq,
    ordination = scaled_wUNI_pcoa,
    color = "lakesite",
    shape = "season",
    title = "Weighted Unifrac PCoA") +
  geom_point(size=5, alpha = 0.5, aes(color = lakesite)) +
  scale_shape_manual(values = c(15,16,17)) +
  scale_color_manual(values = c(lakesite_colors)) + 
  theme_bw()
wUNI_station_pcoa
```

```{r}
# Calculate the BC distance
scaled_wUNI_pcoa <- 
  ordinate(
    physeq = scaled_rooted_physeq,
    method = "PCoA",
    distance = "wunifrac")

# Plot the PCoA
wUNI_station_pcoa <- 
  plot_ordination(
    physeq = scaled_rooted_physeq,
    ordination = scaled_wUNI_pcoa,
    color = "lakesite",
    shape = "limnion",
    title = "Weighted Unifrac PCoA") +
  geom_point(size=5, alpha = 0.5, aes(color = lakesite)) +
  scale_shape_manual(values = c(15,16,17)) +
  scale_color_manual(values = c(lakesite_colors)) + 
  theme_bw()
wUNI_station_pcoa
```


##NMDS

Weighted Unifrac

```{r}
# Calculate the Weighted Unifrac distance
scaled_wUNI_nmds <- 
  ordinate(
    physeq = scaled_rooted_physeq,
    method = "NMDS",
    distance = "wunifrac")
```

```{r}
# Plot the PCoA
wUNI_station_nmds <- 
  plot_ordination(
    physeq = scaled_rooted_physeq,
    ordination = scaled_wUNI_nmds,
    color = "lakesite",
    shape = "nuc_acid_type",
    title = "Weighted Unifrac NMDS") +
  geom_point(size=5, alpha = 0.5, aes(color = lakesite)) +
  scale_shape_manual(values = c(15,16,17)) +
  scale_color_manual(values = c(lakesite_colors)) + 
  theme_bw()
wUNI_station_nmds
```

```{r}
# Plot the PCoA
wUNI_station_nmds <- 
  plot_ordination(
    physeq = scaled_rooted_physeq,
    ordination = scaled_wUNI_nmds,
    color = "lakesite",
    shape = "season",
    title = "Weighted Unifrac NMDS") +
  geom_point(size=5, alpha = 0.5, aes(color = lakesite)) +
  scale_shape_manual(values = c(15,16,17)) +
  scale_color_manual(values = c(lakesite_colors)) + 
  theme_bw()
wUNI_station_nmds
```

```{r}
# Plot the PCoA
wUNI_station_nmds <- 
  plot_ordination(
    physeq = scaled_rooted_physeq,
    ordination = scaled_wUNI_nmds,
    color = "lakesite",
    shape = "limnion",
    title = "Weighted Unifrac NMDS") +
  geom_point(size=5, alpha = 0.5, aes(color = lakesite)) +
  scale_shape_manual(values = c(15,16,17)) +
  scale_color_manual(values = c(lakesite_colors)) + 
  theme_bw()
wUNI_station_nmds
```


##Statistical Significance Testing

PERMANOVA

```{r}
# Calculate all three of the distance matrices
scaled_sorensen_dist <- phyloseq::distance(scaled_rooted_physeq, method = "bray", binary = TRUE)
scaled_bray_dist <- phyloseq::distance(scaled_rooted_physeq, method = "bray")
scaled_wUnifrac_dist <- phyloseq::distance(scaled_rooted_physeq, method = "wunifrac")

# make a data frame from the sample_data
# All distance matrices will be the same metadata because they 
# originate from the same phyloseq object. 
metadata <- data.frame(sample_data(scaled_rooted_physeq))

# Adonis test
# In this example we are testing the hypothesis that the five stations
# that were collected have different centroids in the ordination space 
# for each of the dissimilarity metrics, we are using a discrete variable 
adonis2(scaled_sorensen_dist ~ lakesite, data = metadata)
adonis2(scaled_bray_dist~ lakesite, data = metadata)
adonis2(scaled_wUnifrac_dist~ lakesite, data = metadata)
```

